src/PollController.js

```diff
@@ -42,34 +43,60 @@ function (Okta, FormController) {
       modelEvents: {
         'cancelRequest': '_stopPolling'
       },
-
-      _startPolling: function () {
+      formChildren: [
+        FormType.View({
+          View:
+             '<div >\
+               <div data-se="poll-waiting" class="hide okta-waiting-spinner"></div>\
+             </div>'
+        })
+      ],
+      _doPoll: function () {
         // start polling request 
-        this.transaction.poll()
+        this.transactionObject.poll()
           .then((resp) => {
-            this.options.appState.set('transaction', resp);
-            const factorPollingInterval = resp.factor.profile.refresh;
-            let factorPollingIntervalSeconds = Math.floor(factorPollingInterval/1000);
-            // start one second countdown for next poll request
-            this.countDown  = setInterval(() => {
-              // update title after every second and check if countdown == 1 to poll again
-              const title = Okta.loc('polling.title','login', [factorPollingIntervalSeconds]);
-              this.$el.find('.okta-form-title').text(title);
-              if (factorPollingIntervalSeconds === 1) {
-                // restart poll after countdown hits 1
-                this._stopPolling();
-                this._startPolling();
-              } else {
-                // reduce countdown after every second
-                factorPollingIntervalSeconds = factorPollingIntervalSeconds-1;
-              }
-            }, 1000);
+            if (resp.data && resp.status !== 'POLL') {
+              this.options.appState.set('transaction', resp);
+              return;
+            }
+            this.$('.okta-waiting-spinner').hide();
+            let factorPollingIntervalSeconds = Math.ceil(resp.transaction.profile.refresh/1000);
+            this._startCountDown(factorPollingIntervalSeconds);
           })
           .fail(()=> {
             this._stopPolling();
           });
       },
 
+      _startCountDown: function (factorPollingIntervalSeconds) {
+        // start one second countdown for next poll request
+        this.countDown  = setInterval(() => {
+          // update title after every second and check if countdown == 1 to poll again
+          this._updateTitle(factorPollingIntervalSeconds);
+          if (factorPollingIntervalSeconds === 0) {
+            /* when countdown hits 0
+                         - stop current poll
```

Comment:
nit: Modify comments to say 

   - stop current countDown.
   - show spinner 
    - Check status